### DISCUSSION TITLE
Nuxt 3 Modules Monorepo with `pnpm`; where to ideally put the `playground/`?

### DISCUSSION THREAD

**Discussion by akasection** [nuxt/nuxt#15897] [Questions] [answered]
Hello, I have a case where I have a mono-repository containing collections of Nuxt modules, all managed by pnpm workspace:
```
/my-company-nuxt-monorepo/
  |- packages/
  |--|- some-nuxt-module/
  |--|- another-nuxt-module/
  |- package.json
  |- pnpm-workspace.yaml
```

Usually, when it's a single module repo, by the nuxt recommendations, the `playground/` will be side-by-side with `package.json`, and we can have `pnpm nuxi dev playground` to start the playground.

Now, in this case, where should we put the `playground` folder? is it supposed to be per package, or should it be combined into one `playground/` on root?

or, can we put multiple packages in one playground folder? like:
```
packages/
|- nuxt-a-module/<module.ts>
|- nuxt-b-module/<module.ts>
playground/
|- nuxt-a-module/<nuxt.config.ts>
|- nuxt-b-module/<nuxt.config.ts>
```

**Comment by wobsoriano** [comment]
You have different playgrounds per module?

**Reply by akasection** [reply]
might be yes, or probably I need to unify all of them in one playground to test their compatibility with each other.

So.. the answer is.. mixed.

**Comment by Baroshem** [comment, marked as answer]
I would recommend to have it as a part of the package itself. Having playground for all modules can be missleading for the users that would like to try out your modules (what functionality is actually required for me to try it out?). Same for `docs`. If you are planning to provide documentation for your module, keep it as a part of the package. You can share common deps like nuxt, nuxt/kit, docus for using across all the modules, which is also good because you are also mitigating an issue where one module will work with version X of nuxt and the second will work with version Y.

**Reply by akasection** [reply]
I see.. so I don't have to "mono" the playground, just "mono" the package dependencies. I think I got it. Thanks a lot!
